bundles/org.openhab.io.homekit/src/main/java/org/openhab/io/homekit/internal/HomekitOHItemProxy.java

```diff
@@ -0,0 +1,134 @@
+import org.eclipse.smarthome.core.library.types.HSBType;
+import org.eclipse.smarthome.core.library.types.OnOffType;
+import org.eclipse.smarthome.core.library.types.PercentType;
+import org.eclipse.smarthome.core.types.State;
+import org.eclipse.smarthome.core.types.UnDefType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ *
+ * Proxy class that can collect multiple commands for the same openHAB item and merge them to one command.
+ * e.g. Hue and Saturation update for Color Item
+ * 
+ * @author Eugen Freiter Initial contribution
+ *
+ */
+
+public class HomekitOHItemProxy {
+    private static final Logger logger = LoggerFactory.getLogger(HomekitOHItemProxy.class);
+    public static final String HUE_COMMAND = "hue";
+    public static final String SATURATION_COMMAND = "saturation";
+    public static final String BRIGHTNESS_COMMAND = "brightness";
+    public static final String ON_COMMAND = "on";
+
+    // delay, how long wait for further commands.
+    // TODO: make it configurable ?
+    private final int DELAY = 50;
+
+    private final ScheduledExecutorService scheduler = ThreadPoolManager
+            .getScheduledPool(ThreadPoolManager.THREAD_POOL_NAME_COMMON);
+    private ScheduledFuture<?> future;
+
+    private final Item item;
+
+    @Nullable
+    private ConcurrentHashMap<String, State> commandCache;
+
+    public HomekitOHItemProxy(final Item item) {
+        this.item = item;
+    }
+
+    public Item getItem() {
+        return item;
+    }
+
+    private void sendCommand() {
+        if (item instanceof ColorItem) {
+            final State currentState = item.getState() instanceof UnDefType ? HSBType.BLACK : item.getState();
+            final State hue = commandCache.containsKey(HUE_COMMAND) ? commandCache.remove(HUE_COMMAND)
+                    : ((HSBType) currentState).getHue();
+            final State saturation = commandCache.containsKey(SATURATION_COMMAND)
+                    ? commandCache.remove(SATURATION_COMMAND)
+                    : ((HSBType) currentState).getSaturation();
+            final State brightness = commandCache.containsKey(BRIGHTNESS_COMMAND)
+                    ? commandCache.remove(BRIGHTNESS_COMMAND)
+                    : ((HSBType) currentState).getBrightness();
+            ((ColorItem) item).send(new HSBType((DecimalType) hue, (PercentType) saturation, (PercentType) brightness));
+            logger.trace("send HSB command for item {} with following values h {} s{} b{}", item, hue, saturation,
+                    brightness);
+        } else if (item instanceof DimmerItem) { // not ColorItem but DimmerItem
+            final State on = commandCache.remove(ON_COMMAND);
+            final State brightness = commandCache.remove(BRIGHTNESS_COMMAND);
+
+            // if "ON" command received we dont need to send brightness.
+            // TODO: make brightness suppression on "On" configurable.
+            if (on != null) {
```

Comment:
I'm not 100% sure that is the correct behaviour. Is 100% always send along ON? or can it be 50% and ON? The reason I ask is that I have a lot of KNX dimmers (in fact they are DALI, but connected over KNX). If I select the light in the HomeKit app (state was OFF) and set it to 50% via the slider it always turns on at 100%. IMO this is because ON is interpreted as 50% and overrides the brightness send before. So I think that ON should not be send but the actual brightness if both commands are available.

Reply:
did some tests:
1) Dimmer Off and i click on "device icon": home app sends "On" and Brightness = 100% independent on brightness which was before. 

2) Dimmer On and i change brightness, e.g. from 50% to 60%, home app sends "On" and selected brightness (e.g. 60%). this was not covered by PR. will add commit

3) Dimmer is off and i open "Control" and select brightness, e.g. 60%,  home apps sends "On" and selected brightness. 

to me - we need only to suppress brightness=100% in the first case.

Reply:
Well, at least in my setup „ON“ always results in 100% on KNX dimmers. So I would prefer to suppress the ON instead of the brightness. However, for ZigBee devices this would lead to 100% even if the light was at 40% before and could resume that state.

Reply:
ok. looks like we need to make configurable. 
do i understand correctly we have 2 cases:
- suppress "ON". so, never sends "ON" update only Brightness. 
- suppress 100% brightness (which typically comes with "ON" event)

anything missing ? do we need to suppress "OFF"?

Reply:
The problem is that this would need to be configured on item level. 

What is send along with OFF? 0%? For hue/deconz lights (not sure about zigbee) sending 0% would result in a light going to on after the next ON command but with brightness 0, so actually the light is still off.

In what situation would you want to suppress 100%? All my devices would work perfectly if we only send brightness. Obviously HomeKit wants to turn the light full on if it sends ON and 100%. Or is this a limitation of HAP-Java which generates two commands where HomeKit sends only ON?

Reply:
we had this use case in some tickets/discussion - Dimmer could have  a soft launch on On, i.e. increasing brightness slowly to 100%. 

HomeKit home app is sending always "On" and "brightness", both with one request, e.g. setting brightness with to 72

Reply:
Of course I don't know all devices, but at least all of my device also use some sort of transition time when the brightness changes (i.e. 100% and ON are equal). Also the change from say 10% to 90% is not instantaneous but is slower or faster depending on the configured time for that. 

At least for KNX, DMX, deconz and hue lights this works. I agree that the "double command" is wrong in most cases, but I don't know of any device that would fail or behave wrong in case only the brightness is send. 

Maybe the following would be a good scheme:

ON / 100 % -> ON
ON / 1-99% -> 1-99%
OFF / 0% -> OFF

Would that cover everything?

Reply:
yes, probably your case will cover most of the setups. 
but let's make configurable on the item level.
i add following 4 modes (better names are welcome)
- **none**: nothing is filtered. this is default
- **filterOn**: all "on" events are filtered. only brightness is sent.
- **filterBrightness100**: brightness 100 is filtered when "ON" is sent. everything else with no changes. i.e. "on" along with brightness. this would support "soft launch"
- **filterOnExceptBrightness100**: your proposal. brightness <100% -> "ON" event filtered, brightness = 100% -> only ON is sent

code committed. please take a look.
once we are ok with the logic and naming, i will update README